scale-encode/Cargo.toml

```diff
@@ -14,22 +14,25 @@ keywords.workspace = true
 include.workspace = true
 
 [features]
-default = ["derive", "primitive-types", "bits"]
+default = ["std", "derive", "primitive-types", "bits"]
 
-# Include the derive proc macro
-derive = ["dep:scale-encode-derive"]
+# Activates std feature.
+std = ["scale-info/std"]
 
-# impls for key primitive-types
+# Include the derive proc macro. This is only available in std.
+derive = ["dep:scale-encode-derive", "std"]
+
+# impls for key primitive-types.
 primitive-types = ["dep:primitive-types"]
 
 # impls for Bits.
 bits = ["dep:scale-bits"]
 
 [dependencies]
-scale-info = { version = "2.7.0", features = ["bit-vec"] }
-thiserror = "1.0.37"
-codec = { package = "parity-scale-codec", version = "3.0.0", default-features = false, features = ["derive", "full"] }
-scale-bits = { version = "0.3.0", default-features = false, features = ["scale-info"], optional = true }
+scale-info = { version = "2.7.0",  default-features = false, features = ["bit-vec"] }
+codec = { package = "parity-scale-codec", version = "3.0.0", default-features = false, features = ["derive"] }
+derive_more = { version = "0.99.5" }
```

Comment:
I'd take the same approach as `scale-bits` in this PR; rather than `derive_more`, just custom `Display` and `From` impls for error types and then `#[cfg(feature = "std")] impl std::error::Error for Error {}`. Faster compile times and one less dep; the errors are simple anyway here :)

Reply:
Sure thing 👍